qc-runner/src/cellSizeDistribution.r

```diff
@@ -82,9 +82,22 @@ task <- function(seurat_obj, config, task_name, sample_id) {
     # 3483 6019 3892 3729 4734 3244
     
     plot2_data <- seq_along(plot1_data)
-    plot2_data <- unname(map2(plot1_data,plot2_data,function(x,y){c("u"=x,"rank"=y)}))
+    plot1_data_transformed <- log(plot1_data)
+    plot2_data <- unname(map2(plot1_data_transformed,plot2_data,function(x,y){c("log_u"=x,"rank"=y)}))
     plot1_data <- lapply(unname(plot1_data),function(x) {c("u"=x)})
 
+    # Downsample plotData
+    # Handle when the number of remaining cells is less than the number of cells to downsample
+    num_cells_to_downsample <- min(max(percent_downsample / 100 * ncol(sample_subset), num_cells_to_downsample), ncol(sample_subset))
+          print(paste('num cells to downsample', num_cells_to_downsample))
+      print(paste('num cells in sample', ncol(sample_subset)))
+    set.seed(123)
+    cells_position_to_keep <- sample(1:ncol(sample_subset), num_cells_to_downsample, replace = FALSE)
+    cells_position_to_keep <- sort(cells_position_to_keep)
```

Comment:
Is this sorting necessary?

Reply:
hmm.. I don't think so

Reply:
If it's not sorted, It does rearrange the order of the cells, as sample generates numbers randomly. It may affect the UI, because the cells are not sorted again. I think it's best just to leave it in.